### PR TITLE
Raise a warning if dataset contains files not within BIDS spec

### PR DESCRIPTION
**Pull request by chrisgorgo**
This PR adds a feature warning the user if their datasets contain non-bids compatible files.

### CODE REVIEW

File: validators/bids.js [hunk 1]
```diff
@@ -127,6 +261,18 @@ var BIDS = {
         // validate individual files
         async.forEachOf(fileList, function (file, key, cb) {
             var path = utils.relativePath(file);
+            if (!(self.isTopLevel(path) || self.isSessionLevel(path) || self.isSubjectLevel(path) || self.isAnat(path)
```
Comment (constellates) [inline validators/bids.js:265]: I just tested with ds001 from the new datasets and this warning is being thrown for every file in the dataset.

File: validators/bids.js [hunk 2]
```diff
@@ -115,6 +115,140 @@ var BIDS = {
...
+                return true;
+            }
+        }
+        return false;
+    },
+
+    /**
+     * Check if the file has a name appropriate for a fieldmap scan
+     */
+    isFielMap: function(path) {
```
Comment (constellates) [inline validators/bids.js]: Is this a typo? It's missing the "d" in fieldmap.

**Comment by constellates** [comment]
I also noticed that the syntax change  with the "colors" lib in the CLI at line 10 doesn't work with the string-safe version of colors we're using.  This was a change in the previous merge.

We can either change it back from

```
console.log("This does not appear to be a BIDS dataset. For more info go to http://bids.neuroimaging.io/".red);
```

to

```
console.log(colors.red("This does not appear to be a BIDS dataset. For more info go to http://bids.neuroimaging.io/"));
```

or switch colors out of string safe mode

```
var colors = require('colors/safe');
var colors = require('colors');
```

File: validators/bids.js (hunk 1 above)
Comment (chrisgorgo) [inline validators/bids.js:265]: On the command line? Was there a trailing backslash at the end of the path?

Comment (constellates) [inline validators/bids.js:265]: Here's how I ran it.

```
node bin/bids-validator ~/Desktop/BIDS\ examples/ds001
```

Comment (chrisgorgo) [inline validators/bids.js:265]: There is a bug with paths returned by util.relativePath I need to look
into. When you run it this way:

node bin/bids-validator ~/Desktop/BIDS\ examples/ds001/

It should work

Comment (constellates) [inline validators/bids.js:265]: If I log the path immediately before that check, this I what I get. Is this incorrect?

```
ds001/CHANGES
ds001/README
ds001/dataset_description.json
ds001/participants.tsv
ds001/sub-01/anat/sub-01_T1w.nii.gz
ds001/sub-01/anat/sub-01_inplaneT2.nii.gz
ds001/sub-01/func/sub-01_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-01/func/sub-01_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-01/func/sub-01_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-01/func/sub-01_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-01/func/sub-01_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-01/func/sub-01_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-02/anat/sub-02_T1w.nii.gz
ds001/sub-02/anat/sub-02_inplaneT2.nii.gz
ds001/sub-02/func/sub-02_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-02/func/sub-02_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-02/func/sub-02_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-02/func/sub-02_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-02/func/sub-02_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-02/func/sub-02_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-03/anat/sub-03_T1w.nii.gz
ds001/sub-03/anat/sub-03_inplaneT2.nii.gz
ds001/sub-03/func/sub-03_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-03/func/sub-03_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-03/func/sub-03_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-03/func/sub-03_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-03/func/sub-03_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-03/func/sub-03_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-04/anat/sub-04_T1w.nii.gz
ds001/sub-04/anat/sub-04_inplaneT2.nii.gz
ds001/sub-04/func/sub-04_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-04/func/sub-04_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-04/func/sub-04_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-04/func/sub-04_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-04/func/sub-04_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-04/func/sub-04_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-05/anat/sub-05_T1w.nii.gz
ds001/sub-05/anat/sub-05_inplaneT2.nii.gz
ds001/sub-05/func/sub-05_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-05/func/sub-05_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-05/func/sub-05_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-05/func/sub-05_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-05/func/sub-05_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-05/func/sub-05_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-06/anat/sub-06_T1w.nii.gz
ds001/sub-06/anat/sub-06_inplaneT2.nii.gz
ds001/sub-06/func/sub-06_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-06/func/sub-06_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-06/func/sub-06_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-06/func/sub-06_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-06/func/sub-06_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-06/func/sub-06_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-07/anat/sub-07_T1w.nii.gz
ds001/sub-07/anat/sub-07_inplaneT2.nii.gz
ds001/sub-07/func/sub-07_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-07/func/sub-07_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-07/func/sub-07_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-07/func/sub-07_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-07/func/sub-07_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-07/func/sub-07_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-08/anat/sub-08_T1w.nii.gz
ds001/sub-08/anat/sub-08_inplaneT2.nii.gz
ds001/sub-08/func/sub-08_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-08/func/sub-08_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-08/func/sub-08_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-08/func/sub-08_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-08/func/sub-08_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-08/func/sub-08_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-09/anat/sub-09_T1w.nii.gz
ds001/sub-09/anat/sub-09_inplaneT2.nii.gz
ds001/sub-09/func/sub-09_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-09/func/sub-09_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-09/func/sub-09_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-09/func/sub-09_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-09/func/sub-09_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-09/func/sub-09_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-10/anat/sub-10_T1w.nii.gz
ds001/sub-10/anat/sub-10_inplaneT2.nii.gz
ds001/sub-10/func/sub-10_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-10/func/sub-10_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-10/func/sub-10_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-10/func/sub-10_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-10/func/sub-10_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-10/func/sub-10_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-11/anat/sub-11_T1w.nii.gz
ds001/sub-11/anat/sub-11_inplaneT2.nii.gz
ds001/sub-11/func/sub-11_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-11/func/sub-11_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-11/func/sub-11_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-11/func/sub-11_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-11/func/sub-11_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-11/func/sub-11_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-12/anat/sub-12_T1w.nii.gz
ds001/sub-12/anat/sub-12_inplaneT2.nii.gz
ds001/sub-12/func/sub-12_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-12/func/sub-12_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-12/func/sub-12_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-12/func/sub-12_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-12/func/sub-12_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-12/func/sub-12_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-13/anat/sub-13_T1w.nii.gz
ds001/sub-13/anat/sub-13_inplaneT2.nii.gz
ds001/sub-13/func/sub-13_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-13/func/sub-13_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-13/func/sub-13_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-13/func/sub-13_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-13/func/sub-13_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-13/func/sub-13_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-14/anat/sub-14_T1w.nii.gz
ds001/sub-14/anat/sub-14_inplaneT2.nii.gz
ds001/sub-14/func/sub-14_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-14/func/sub-14_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-14/func/sub-14_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-14/func/sub-14_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-14/func/sub-14_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-14/func/sub-14_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-15/anat/sub-15_T1w.nii.gz
ds001/sub-15/anat/sub-15_inplaneT2.nii.gz
ds001/sub-15/func/sub-15_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-15/func/sub-15_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-15/func/sub-15_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-15/func/sub-15_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-15/func/sub-15_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-15/func/sub-15_task-balloonanalogrisktask_run-03_events.tsv
ds001/sub-16/anat/sub-16_T1w.nii.gz
ds001/sub-16/anat/sub-16_inplaneT2.nii.gz
ds001/sub-16/func/sub-16_task-balloonanalogrisktask_run-01_bold.nii.gz
ds001/sub-16/func/sub-16_task-balloonanalogrisktask_run-01_events.tsv
ds001/sub-16/func/sub-16_task-balloonanalogrisktask_run-02_bold.nii.gz
ds001/sub-16/func/sub-16_task-balloonanalogrisktask_run-02_events.tsv
ds001/sub-16/func/sub-16_task-balloonanalogrisktask_run-03_bold.nii.gz
ds001/sub-16/func/sub-16_task-balloonanalogrisktask_run-03_events.tsv
ds001/task-balloonanalogrisktask_bold.json
```

**Comment by chrisgorgo** [comment]
Thank you for a through review. I fixed the issues you have outlined. Could you have another look?

**Comment by constellates** [comment]
All those fixes look good, but I noticed one more small issue in the CLI. 

The final success console.log needs to be in a conditional. Currently it get's logged regardless of whether there are errors or not. 

Something along the lines of this

```
validate.BIDS(dir, function (errors, warnings) {
    if (errors === 'Invalid') {
        console.log(colors.red("This does not appear to be a BIDS dataset. For more info go to http://bids.neuroimaging.io/"));
    } else {
        logIssues(errors, 'red');
        logIssues(warnings, 'yellow');
    }
    console.log(colors.green("This dataset appears to be BIDS compatible."));
});


validate.BIDS(dir, function (errors, warnings) {
    if (errors === 'Invalid') {
        console.log(colors.red("This does not appear to be a BIDS dataset. For more info go to http://bids.neuroimaging.io/"));
    } else if (errors.length > 0 || warnings.length > 0) {
        logIssues(errors, 'red');
        logIssues(warnings, 'yellow');
    } else {
        console.log(colors.green("This dataset appears to be BIDS compatible."));
    }
});
```

**Comment by chrisgorgo** [comment]
Good catch! This should be fixed now!

**Pull request merged**